File: backend/Billing.Api/Entities/Invoice.cs
using Billing.Api.Enums;

namespace Billing.Api.Entities;

public class Invoice
{
    public Guid Id { get; set; }

    public string Number { get; set; } = string.Empty;

    public InvoiceStatus Status { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime? ClosedAt { get; set; }
}

File: frontend/src/app/core/models/invoice.ts
export type InvoiceStatus = 0 | 1;

export interface Invoice {
  id: string;
  number: string;
  status: InvoiceStatus;
  createdAt: string;
  closedAt: string | null;
}

export interface ApiErrorResponse {
  statusCode: number;
  message: string;
  traceId: string;
}
